### PR TITLE
Components: Text - Slight refactor and exports as experimental

### LINKED ISSUE #21084: Components: (Experimental) Text Component is not exported
Follow up to https://github.com/WordPress/gutenberg/pull/18495.
The (experimental) `Text` component is not exported.

### PR DESCRIPTION
**Pull request by ItsJonQ**
## Description
This update refactors the (experimental) `Text` primitive and exports the component as `__experimentalText` from `@wordpress/components`

## How has this been tested?
* Tested in local Storybook
* Tested in Jest/JSDOM

## Types of changes
* Refactor styles and file format
* Exports `Text` component as `__experimentalText` in `components/src/index.js`
* Adds unit tests
* Update README file

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR. 

Resolves: https://github.com/WordPress/gutenberg/issues/21084

### CODE REVIEW

File: packages/components/src/index.js [hunk 1]
```diff
@@ -80,6 +80,7 @@ export { default as Snackbar } from './snackbar';
 export { default as SnackbarList } from './snackbar/list';
 export { default as Spinner } from './spinner';
 export { default as TabPanel } from './tab-panel';
+export { default as __experimentalText } from './text';
```
Comment (gziolo) [inline packages/components/src/index.js]: There is

`export * from './text';`

in the last line.

Comment (ItsJonQ) [reply packages/components/src/index.js]: Oh! I missed that. I'll update that.
I think this update feels more inline with how we've been exporting our other components 🙏

Comment (gziolo) [reply packages/components/src/index.js]: I agree that the proposed changes make sense 👍

**Comment by ItsJonQ** [comment]
@gziolo I've merged in the latest `master` and adjusted my PR to fit the latest `Text` [RNMobile updates](https://github.com/WordPress/gutenberg/pull/20745)

cc'ing @SergioEstevao to double check (👋 hai!!!)

Thank you!

**Review by SergioEstevao — APPROVED** [review]
I tested the refactor on GB mobile and it's working fine.

**Pull request merged**